core/src/main/java/edu/wpi/grip/core/Main.java

```diff
@@ -1,6 +1,38 @@
 package edu.wpi.grip.core;
 
+import com.google.common.eventbus.EventBus;
+import edu.wpi.grip.core.operations.Operations;
+import edu.wpi.grip.core.serialization.Project;
+import edu.wpi.grip.generated.CVOperations;
+
+import java.io.File;
+
+/**
+ * Main driver class for headless mode
+ */
 public class Main {
-    public static void main(String[] args) {
+    public static void main(String[] args) throws Exception {
+        if (args.length != 1) {
+            System.err.println("Usage: GRIP.jar project.grip");
+            return;
+        }
+
+        final String projectPath = args[0];
+
+        final EventBus eventBus = new EventBus();
+        final Pipeline pipeline = new Pipeline(eventBus);
+        final Palette palette = new Palette(eventBus);
+        final Project project = new Project(eventBus, pipeline, palette);
```

Comment:
We should have one way of creating these objects and the UI should take advantage of it.
Do something like `main.setup()`.
That way if we change any of the constructors we don't need to worry about it.

As for dependency injection, lets take care of that when it actually gets implemented.